pykokkos/core/translators/bindings.py

```diff
@@ -536,9 +537,23 @@ def bind_main_single(
 
     main: List[str] = translate_mains(source, functor, members, pk_import)
     params: Dict[str, str] = get_kernel_params(members, False, True, real)
+
+    # fall back to the old hard-coded default
+    # for now--this includes cases where an
+    # accumulator is not even defined
+    acc_type = "double"
+
+    for element in source[0]:
+        # TODO: support more types
+        if "pk.Acc" in element:
+            if "pk.int64" in element:
+                acc_type = "int64_t"
+            elif "pk.double" in element:
+                acc_type = "double"
+
     signature: str = generate_kernel_signature("void", kernel_name, params)
     instantiation: str = generate_functor_instance(functor, members)
-    acc: str = f"double {Keywords.Accumulator.value} = 0;"
+    acc: str = f"{acc_type} {Keywords.Accumulator.value} = 0;"
```

Comment:
@NaderAlAwar The diff should be a bit cleaner to review now. AFAICT this change makes it possible mutate the `type` of the Accumulator, which it seems was previously hardcoded.

While the previous discussion focused on making it possible to change the reduction result view type to match the accumulator type, it seems that wouldn't have much value if the accumulator type is hardcoded anyway, so this is at least a first step that allows more of the new test cases that I wrote to pass compared with `develop`, but not all of them yet.